spring-web/src/test/java/org/springframework/http/client/reactive/ClientHttpConnectorTests.java

```diff
@@ -172,6 +180,70 @@ void cancelResponseBody(ClientHttpConnector connector) {
 				.verify();
 	}
 
+	@ParameterizedConnectorTest
+	void testExpiresMaxAgeAttributes(ClientHttpConnector connector) {
+		// maxAge is set to 12 days from system time.
+		long maxAge = 1036800L;
+
+		// expires date is set to 10 days from system time.
+		long expires = 864000L;
+		ZonedDateTime currentDateTime = ZonedDateTime.now(java.time.ZoneOffset.UTC);
+		ZonedDateTime futureDateTime = currentDateTime.plusSeconds(expires);
+
+		// processing time may affect the calculation of ZonedDateTime.now during merge of expires and max age
+		// therefore we check range with buffer of 2 seconds
+		long maxAgeLowerLimit = maxAge - 2;
+		long maxAgeUpperLimit = maxAge + 2;
+		long expiresLowerLimit = expires - 2;
+		long expiresUpperLimit = expires + 2;
+
+		List<HttpCookie> httpCookies = getHttpCookies(maxAge, futureDateTime);
+		prepareResponse(response -> {
+			response.setResponseCode(200);
+			httpCookies.forEach(httpCookie -> response.addHeader("Set-Cookie", httpCookie.toString()));
+		});
+
+		ClientHttpResponse response = connector.connect(HttpMethod.POST, this.server.url("/").uri(),
+				ReactiveHttpOutputMessage::setComplete).block();
+		assertThat(response).isNotNull();
+		assertThat(response.getCookies()).isNotEmpty();
+
+		List<ResponseCookie> maxAgeAndExpiresCookies = response.getCookies().get(MAX_AGE_AND_EXPIRES);
+		assertThat(maxAgeAndExpiresCookies).size().isEqualTo(1);
+		Duration maxAgeAndExpires = maxAgeAndExpiresCookies.get(0).getMaxAge();
+		assertThat(maxAgeAndExpires.getSeconds()).isGreaterThanOrEqualTo(maxAgeLowerLimit);
+		assertThat(maxAgeAndExpires.getSeconds()).isLessThanOrEqualTo(maxAgeUpperLimit);
+
+		List<ResponseCookie> maxAgeOnlyCookies = response.getCookies().get(MAX_AGE_ONLY);
+		assertThat(maxAgeOnlyCookies).size().isEqualTo(1);
+		Duration maxAgeOnly = maxAgeOnlyCookies.get(0).getMaxAge();
+		assertThat(maxAgeOnly.getSeconds()).isGreaterThanOrEqualTo(maxAgeLowerLimit);
+		assertThat(maxAgeOnly.getSeconds()).isLessThanOrEqualTo(maxAgeUpperLimit);
+
+		List<ResponseCookie> expiresOnlyCookies = response.getCookies().get(EXPIRES_ONLY);
+		assertThat(expiresOnlyCookies).size().isEqualTo(1);
+		Duration expiresOnly = expiresOnlyCookies.get(0).getMaxAge();
+		assertThat(expiresOnly.getSeconds()).isGreaterThanOrEqualTo(expiresLowerLimit);
+		assertThat(expiresOnly.getSeconds()).isLessThanOrEqualTo(expiresUpperLimit);
+	}
+
+	private List<HttpCookie> getHttpCookies(long maxAge, ZonedDateTime futureDateTime) {
+		String expires = futureDateTime.format(DateTimeFormatter.RFC_1123_DATE_TIME);
```

Comment:
Consider adding input validation for the maxAge parameter (e.g., check if it's positive) and the futureDateTime (e.g., check if it's in the future). 
something like this below
if (maxAge < 0) {
        throw new IllegalArgumentException("maxAge must be non-negative");
    }
    if (futureDateTime.isBefore(ZonedDateTime.now())) {
        throw new IllegalArgumentException("futureDateTime must be in the future");
    }

Reply:
This is a helper method in a test class!!

If someone wants to reuse this method in their tests and pass invalid arguments, their test assertion has to handle http response accordingly.

Reply:
ok